src/db/connection.cr

```diff
@@ -21,10 +21,12 @@ module DB
   abstract class Connection
     include Disposable
     include SessionMethods(Connection, Statement)
+    include BeginTransaction
 
     # :nodoc:
     getter database
     @statements_cache = StringKeyCache(Statement).new
+    @transaction = false
```

Comment:
It was simpler to just store the bool.
We can have a `TopLevelTransaction` or `WeakRef(TopLevelTransaction)` but if doing that we should deal with explicit management of what will happen when the connection or parent transaction is closed IMO.